src/content/docs/tutorials/rust/social_network.mdx

```diff
@@ -0,0 +1,222 @@
+    for row in result {
+        let row_str = row.iter()
+            .map(|col| col.to_string())  // Convert each column to a string
+            .collect::<Vec<String>>()    // Collect into a Vec<String>
+            .join(", ");                 // Join elements with ", "
+    
+        println!("{}", row_str);
+    } 
+```
+Returning:
+```
+darkdog878, 6
+```
+
+### Q2: What is the shortest path between two users?
+Another question which we might be interested in answering is finding the shortest path between two users. For example, how many follows does it take to get from user `silentguy245` to `epicwolf202`?
+1. We may start off the query with a simple `MATCH` query followed by a `WHERE` query to specify the results we want:
+```Rust
+    conn.query("""
+        MATCH p = (u1:user)-[f:FOLLOWS]->(u2:User)
+        WHERE u1.username = 'silentguy245' AND u2.username = 'epicwolf202'
+        RETURN p;
+    """)?;
+```
+This will return us whether `silentguy245` follows `epicwolf202`, which they don't. 
+2. We can then build off this query using [recursive match](../../cypher/query-clauses/match.md) to find path instead:
+```Rust
+    conn.query("""
+        MATCH p = (u1:user)-[f:FOLLOWS* SHORTEST 1..4]->(u2:User)
+        WHERE u1.username = 'silentguy245' AND u2.username = 'epicwolf202'
+        RETURN p;
+    """)?;
+```
+This will return us a result in `RECURSIVE_REL` datatype, which is not as useful to interpret.
+3. We will then simplify the result using some [recursive relationship functions](../../cypher/expressions/recursive-rel-functions.md)
+```Rust
+    conn.query("""
+        MATCH p = (u1:user)-[f:FOLLOWS* SHORTEST 1..4]->(u2:User)
+        WHERE u1.username = 'silentguy245' AND u2.username = 'epicwolf202'
+        RETURN properties(nodes(p), 'username');
+    """)?;
+```
+
+### Q3: How many 3-hop paths exist between userA and userB that passes through userC?
+To answer this query, we should do something similar to Q1. We first match all 3-hop queries:
+```Rust
+    conn.query("""
+        MATCH (u1:User)-[f1:FOLLOWS]->(u2:User)-[f2:Follows]->(u3:User)-[f3:FOLLOWS]->(u4:User)
+        RETURN count(u4);
+    """)?;
+```
+Next, we add to the query to match only cases concerning with userA, userB, and userC. In this case, we want to introduce the idea of perpared statements, so that we can easily swap out elements of the query for other users. We arbitrarily choose userA to be `epicwolf202`, userB to be `stormcat597`, and userC to be `stormfox762`
+```Rust
+    let u1 = "epicwolf202";
+    let u2 = "stormcat597";
+    let u3 = "stormfox762";
+    let query = format!("""
+        MATCH (u1:User)-[f1:FOLLOWS]->(u2:User)-[f2:Follows]->(u3:User)-[f3:FOLLOWS]->(u4:User)
+        WHERE u1.username= {}, AND u4.username= {} AND (u2.username= {} OR u3.username= {})
+        RETURN count(u4);
+    """, u1, u2, u3, u3);
+    conn.query(query)?;
+```
+
```

Comment:
Can we link to a single `main.rs` file that the user can just copy paste entirely? So that when they read through to this point, they can just see the entire file and copy-paste the entire code in one go.